test/unit/math/mix/fun/offset_multiplier_constrain_helpers.hpp

```diff
@@ -0,0 +1,65 @@
+#ifndef STAN_TEST_UNIT_MATH_MIX_FUN_offset_multiplier_CONSTRAIN_HELPERS_HPP
+#define STAN_TEST_UNIT_MATH_MIX_FUN_offset_multiplier_CONSTRAIN_HELPERS_HPP
+
+namespace offset_multiplier_constrain_tests {
+template <typename T1, typename T2, typename T3>
+void expect(const T1& x, const T2& mu, const T3& sigma) {
+  auto f1 = [](const auto& x, const auto& mu, const auto& sigma) {
+    return stan::math::offset_multiplier_constrain(x, mu, sigma);
+  };
+  auto f2 = [](const auto& x, const auto& mu, const auto& sigma) {
+    stan::return_type_t<decltype(x), decltype(mu), decltype(sigma)> lp = 0;
+    return stan::math::offset_multiplier_constrain(x, mu, sigma, lp);
+  };
+  auto f3 = [](const auto& x, const auto& mu, const auto& sigma) {
+    stan::return_type_t<decltype(x), decltype(mu), decltype(sigma)> lp = 0;
+    stan::math::offset_multiplier_constrain(x, mu, sigma, lp);
+    return lp;
+  };
+  auto f4 = [](const auto& x, const auto& mu, const auto& sigma) {
+    using lub_t
+        = stan::return_type_t<decltype(x), decltype(mu), decltype(sigma)>;
+    lub_t lp = 0;
+    auto xx = stan::math::offset_multiplier_constrain(x, mu, sigma, lp);
+    return stan::math::add(lp, lub_t(stan::math::sum(xx)));
+  };
+
+  stan::test::expect_ad(f1, x, mu, sigma);
+  stan::test::expect_ad(f2, x, mu, sigma);
+  stan::test::expect_ad(f3, x, mu, sigma);
+  stan::test::expect_ad(f4, x, mu, sigma);
+}
+template <typename T1, typename T2, typename T3>
+void expect_vec(const T1& x, const T2& mu, const T3& sigma) {
+  auto f1 = [](const auto& x, const auto& mu, const auto& sigma) {
+    return stan::math::offset_multiplier_constrain(x, mu, sigma);
+  };
+  auto f2 = [](const auto& x, const auto& mu, const auto& sigma) {
+    stan::return_type_t<decltype(x), decltype(mu), decltype(sigma)> lp = 0;
+    return stan::math::offset_multiplier_constrain(x, mu, sigma, lp);
+  };
+  auto f3 = [](const auto& x, const auto& mu, const auto& sigma) {
+    stan::return_type_t<decltype(x), decltype(mu), decltype(sigma)> lp = 0;
+    stan::math::offset_multiplier_constrain(x, mu, sigma, lp);
+    return lp;
+  };
+  auto f4 = [](const auto& x, const auto& mu, const auto& sigma) {
```

Comment:
I don't think this `f4` tests anything that the other things don't cover. Given the test sizes here are problematic, I want to delete it. Is this okay or am I missing something?

Reply:
test 4 just takes the gradient wrt lp and the output. I found it useful when doing the tests for lub since it has a complex gradient. But I'm fine with taking them in or out